prow/plugins/verify-owners/verify-owners.go

```diff
@@ -0,0 +1,188 @@
+		}
+	}()
+	if err := r.CheckoutPullRequest(pre.Number); err != nil {
+		return err
+	}
+
+	// Check each OWNERS file.
+	for _, f := range modifiedOwnersFiles {
+		// Try to load OWNERS file.
+		path := filepath.Join(r.Dir, f)
+		b, err := ioutil.ReadFile(path)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to read %s.", path)
+			return nil
+		}
+		var approvers []string
+		var labels []string
+		simple, err := repoowners.ParseSimpleConfig(b)
+		if err != nil || simple.Empty() {
+			full, err := repoowners.ParseFullConfig(b)
+			if err != nil {
+				wrongOwnersFiles[f] = fmt.Errorf("error occurred parsing the OWNERS file in %s: %v", f, err)
+				continue
+			} else {
+				// it's a FullConfig
+				for _, config := range full.Filters {
+					approvers = append(approvers, config.Approvers...)
+					labels = append(labels, config.Labels...)
+				}
+			}
+		} else {
+			// it's a SimpleConfig
+			approvers = simple.Config.Approvers
+			labels = simple.Config.Labels
+		}
+		// Check labels against blacklist
+		if sets.NewString(labels...).HasAny(labelsBlackList...) {
+			wrongOwnersFiles[f] = fmt.Errorf("OWNERS file contains blacklisted labels: %s", sets.NewString(simple.Config.Labels...).Intersection(sets.NewString(labelsBlackList...)).List())
+			continue
+		}
+		// Check approvers isn't empty
+		if filepath.Dir(f) == "." && len(approvers) == 0 {
+			wrongOwnersFiles[f] = errors.New("no approvers in the root directory OWNERS file")
+			continue
+		}
+	}
+	// React if we saw something.
+	if len(wrongOwnersFiles) > 0 {
+		if err := ghc.AddLabel(org, repo, pre.Number, invalidOwnersLabel); err != nil {
+			return err
+		}
+		log.Debugf("Creating a review for %d OWNERS files.", len(wrongOwnersFiles))
+		var comments []github.DraftReviewComment
+		for errFile, errMsg := range wrongOwnersFiles {
+			resp := fmt.Sprintf("Invalid OWNERS file: %v", errMsg)
+			comments = append(comments, github.DraftReviewComment{
+				Path: errFile,
+				Body: resp,
+			})
+		}
+		err := ghc.CreateReview(org, repo, pre.Number, github.DraftReview{
+			Action:   github.Comment,
+			Comments: comments,
+		})
+		if err != nil {
+			return fmt.Errorf("error creating a review for invalid OWNERS file(s): %v", err)
+		}
+	} else {
+		// Don't bother checking if it has the label...it's a race, and we'll have
+		// to handle failure due to not being labeled anyway.
+		labelNotFound := true
+		if err := ghc.RemoveLabel(org, repo, pre.Number, invalidOwnersLabel); err != nil {
+			if _, labelNotFound = err.(*github.LabelNotFound); !labelNotFound {
+				return fmt.Errorf("failed removing lgtm label: %v", err)
+			}
+			// If the error is indeed *github.LabelNotFound, consider it a success.
+		}
+		// Creates a comment to inform participants that LGTM label is removed due to new
+		// pull request changes.
+		if !labelNotFound {
+			return ghc.CreateComment(org, repo, pre.Number, "OWNERS files are correct. invalid-OWNERS label has been removed.")
+		}
```

Comment:
This last if statement isn't necessary and will just spam the participants on the PR. We only leave comments when necessary to avoid filling PRs with bot comments.
If we are reaching this case the user must have changed their PR to fix the invalid OWNERS file(s) so it will be apparent why the invalid OWNERS label is removed. We do this in the LGTM plugin because it isn't clear to users why `lgtm` is automatically removed by the bot and because it prevents merge whereas removing the invalid OWNERS label will move towards allowing merge.